submissions/ik-web/frogger/app.js

```diff
@@ -0,0 +1,198 @@
+
+// Generate enemies' speed
+function enemySpeed() {
+    return Math.floor( Math.random() * (enemySpeedMax - enemySpeedMin) + enemySpeedMin);
+}
+
+// Add a new enemy after level up
+function addEnemy() {
+    let enemy = new Enemy(enemyPosX, enemyPosY(), enemySprite, enemySpeed(), player);
+    allEnemies.push(enemy);
+}
+
+// Clear enemies if the player won or lose
+function clearEnemy() {
+    allEnemies = [];
+    addEnemy();
+}
+
+//Create level counter
+function addLevelCounter() {
+    levelCounter.style.cssText = `
+        position: fixed; 
+        top: 10px; 
+        left: 50%;
+        font-family: sans-serif;
+        font-size: 30px; 
+        transform: translateX(-50%);
+    `;
+
+    levelCounter.innerHTML = `Level: ${playerStartLevel}`;
+    document.body.prepend(levelCounter);
+}
+
+//----------------------------------------------
+// Constructor for create character
+const Character = function(x, y, sprite) {
+    this.x = x;
+    this.y = y;
+    this.sprite = sprite;
+};
+
+// Method for character update
+Character.prototype.update = function() {};
+
+// Draw the character on the screen, required method for game
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+//----------------------------------------------
+// Constructor for create enemy
+const Enemy = function(x, y, sprite, speed, player) {
+    Character.call(this, x, y, sprite);
+    this.speed = speed;
+};
+
+Enemy.prototype = Object.create(Character.prototype);
+
+//Check characters collision
+Enemy.prototype.enemyCollision = function() {
+    if (this.x - borderCollision <= player.x        // Left
+        && this.y - borderCollision <= player.y     // Up
+        && this.x + borderCollision >= player.x     // Right
+        && this.y + borderCollision >= player.y) {  // Down
+        player.x = playerStartPosX;
+        player.y = playerStartPosY;
+
+        setTimeout(function() {
+            alert(`Game over...\nYou received ${player.lvl} level.`);
+            player.lvl = 1;
+            clearEnemy();
+            levelCounter.innerHTML = `Level: 1`;
+        }, 100);
+    }
+};
+
+// Method for enemies update
+Enemy.prototype.update = function(dt) {
+    this.enemyCollision();
+
+    if (this.x > areaWidth) {
+        this.x = enemyPosX;
+        this.speed = enemySpeed();
+    } else {
+        this.x += this.speed * dt;
+    }
+};
+
+//----------------------------------------------
+// Constructor for create player
+const Player = function(x, y, sprite) {
+    Character.call(this, x, y, sprite);
+    this.lvl = playerStartLevel;
+};
+
+Player.prototype = Object.create(Character.prototype);
+
+//Method for checking game progress
+Player.prototype.checkProgress = function() {
+    if (this.y < 0) {
+
+        setTimeout(() => {
+            this.x = playerStartPosX;
+            this.y = playerStartPosY;
+            this.lvl++;
+            levelCounter.innerHTML = `Level: ${this.lvl}`;
```

Comment:
`levelCounter` innerHTML is assigned in multiple places and with different approaches. Introduce and use a function for this repetetive operation. This will reduce code repetition and will make code more maintainable.

Reply:
I fixed it

Reply:
Does the function adds score or sets/assigns score? Is it score or level? Some of the parameters att call times say `lvl` (i assume this abbreviation means 'level')

Reply:
I fixed it. Thanks for this useful information